### PR TITLE
Resubmit GH #14

### LINKED ISSUE #221: ensure /quit_server calls for full HTTP/1.1 keepalive compatibility
creaktive@gmail.com reported on Mar 16, 2013 

<pre>
(also issued at <a href="http://web.archive.org/web/20130621091334/https://github.com/libwww-perl/WWW-Mechanize/pull/14" rel="nofollow">https://github.com/libwww-perl/WWW-Mechanize/pull/14</a>)

The t/local/encoding.t &amp; t/local/referer.t tests are potentially deadlockable:

PERL5OPT=-MLWP::Protocol::Net::Curl=http_version,2 prove -lv t/local/encoding.t t/local/referer.t

Here, the testing server waits for the client to close connection, and vice-versa. The client could be forced to use a non-keepalive connection, which solves the deadlock:

PERL5OPT=-MLWP::Protocol::Net::Curl=http_version,1 prove -lv t/local/encoding.t t/local/referer.t

Both tests could also include an explicit undef $agent. The attached code uses the already-available /quit_server URI to do the things in a safest way.
</pre>

### Details

Imported from [Google Code issue 244](http://code.google.com/p/www-mechanize/issues/detail?id=244) via [archive](https://web.archive.org/web/20150227111857/http://code.google.com/p/www-mechanize/issues/detail?id=244)

* **Type**:      Defect
* **Date**:      Mar 16, 2013
* **Reporter**:  creaktive@gmail.com
* **Owner**:     ----
* **Priority**:  Medium
* **Status**:    New

### PR DESCRIPTION
**Pull request by simbabque**
I've rebased #14, tested and added a workflow job to run a regression test. Happy to discuss if we really need it, or if this is such an unusual edge case that it makes no sense to run this every time.

This closes #14 and fixes #221.

### CODE REVIEW

File: .github/workflows/build-and-test.yml [hunk 1]
```diff
@@ -31,120 +31,139 @@ jobs:
...
+  #   steps:
+  #     - uses: actions/checkout@v2 # codecov wants to be inside a Git repository
+  #     - uses: actions/download-artifact@master
+  #       with:
+  #         name: build_dir
+  #         path: .
+  #     - name: Install deps and test
+  #       run: cpan-install-dist-deps --with-develop && test-dist
+  #       env:
+  #         CODECOV_TOKEN: ${{secrets.CODECOV_TOKEN}}
```
Comment (oalders) [inline .github/workflows/build-and-test.yml:49]: Why is the coverage job disabled?

File: .github/workflows/build-and-test.yml [hunk 2]
```diff
@@ -31,120 +31,139 @@ jobs:
...
+  http-one-one-regression-job:
+    name: HTTP/1.1 Keepalive Regression Test - see GH \# 14
     needs: build-job
     runs-on: ubuntu-latest
     container:
       image: perldocker/perl-tester:5.34
     steps:
-      - uses: actions/checkout@v2 # codecov wants to be inside a Git repository
       - uses: actions/download-artifact@master
```
Comment (oalders) [inline .github/workflows/build-and-test.yml:57]: This should probably use `@v3`

File: .github/workflows/build-and-test.yml [hunk 3]
```diff
@@ -31,120 +31,139 @@ jobs:
...
+  #         - "5.16"
+  #         - "5.18"
+  #         - "5.20"
+  #         - "5.22"
+  #         - "5.24"
+  #         - "5.26"
+  #         - "5.28"
+  #         - "5.30"
+  #         - "5.32"
+  #         - "5.34"
```
Comment (oalders) [inline .github/workflows/build-and-test.yml:91]: Could add 5.36 here.

File: .github/workflows/build-and-test.yml [hunk 4]
```diff
@@ -31,120 +31,139 @@ jobs:
...
+  #   steps:
+  #     - uses: actions/checkout@v2
+  #     - uses: actions/download-artifact@master
```
Comment (oalders) [inline .github/workflows/build-and-test.yml:94]: These should both probably use `@v3`

File: .github/workflows/build-and-test.yml [hunk 5]
```diff
@@ -31,120 +31,139 @@ jobs:
...
+  #         - "5.16"
+  #         - "5.18"
+  #         - "5.20"
+  #         - "5.22"
+  #         - "5.24"
+  #         - "5.26"
+  #         - "5.28"
+  #         - "5.30"
+  #         - "5.32"
+  #         - "5.34"
```
Comment (oalders) [inline .github/workflows/build-and-test.yml:120]: Could add 5.36 here.

File: .github/workflows/build-and-test.yml [hunk 6]
```diff
@@ -31,120 +31,139 @@ jobs:
...
+  #         - "5.28"
+  #         - "5.30"
+  #         - "5.32"
+  #   steps:
+  #     - name: Set Up Perl
+  #       uses: shogo82148/actions-setup-perl@v1
+  #       with:
+  #         perl-version: ${{ matrix.perl-version }}
+  #         distribution: strawberry # this option only used on Windows
+  #     - uses: actions/download-artifact@master
```
Comment (oalders) [inline .github/workflows/build-and-test.yml:160]: This should probably use `@v3`

**Review by oalders — CHANGES_REQUESTED** [review]
Overall it looks good, but I think there are some disabled jobs that need to be re-enabled?

**Comment by simbabque** [comment]
> Overall it looks good, but I think there are some disabled jobs that need to be re-enabled?

Oh no. That must have been the heat. I re-enabled them, and force-pushed but forgot to add the file first. Sorry about that. It's up to date now and building.

I'm going to bump the other suggestions in a separate PR.

**Review by oalders — APPROVED** [review]
🎊
